Sources/Controllers/2.0/OAEditFavoriteViewController.mm

```diff
@@ -966,32 +1073,32 @@ - (void)multilineTextField:(id<MDCMultilineTextInput> _Nonnull)multilineTextFiel
 
 #pragma mark - OAPoiTableViewCellDelegate
 
-- (void) onPoiCategorySelected:(NSString *)category
+- (void) onPoiCategorySelected:(NSString *)category index:(NSInteger)index
 {
     _selectedIconCategoryName = category;
+    _selectedCategoryIndex = index;
     [self generateData];
-    [self.tableView reloadData];
-    [self.tableView layoutSubviews];
+    OAPoiTableViewCell *cell = [self.tableView cellForRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:2]];
+    [self.tableView beginUpdates];
```

Comment:
what purpose of beginUpdates and endUpdates? Is reloadRowsAtIndexPaths not enough?